cores/esp8266/core_esp8266_si2c.c

```diff
@@ -65,22 +141,47 @@ void twi_setClockStretchLimit(uint32_t limit){
   twi_clockStretchLimit = limit * TWI_CLOCK_STRETCH_MULTIPLIER;
 }
 
-void twi_init(unsigned char sda, unsigned char scl){
+void twi_init(unsigned char sda, unsigned char scl)
+{
+  // set timer function 
+  ets_timer_setfn(&timer, onTimer, NULL);
+  
+  // create event task
+  ets_task(eventTask, EVENTTASK_QUEUE_PRIO, eventTaskQueue, EVENTTASK_QUEUE_SIZE);
```

Comment:
When using I2C in master mode, it should not be necessary to create this event task and the timer. Same applies to twi_txBuffer and twi_rxBuffer — these should be linked in if only master mode is used.